astropy/io/fits/tests/test_core.py

```diff
@@ -593,6 +595,36 @@ def test_open_nonexistent(self):
             assert os.path.exists(self.temp('foobar.fits'))
             os.remove(self.temp('foobar.fits'))
 
+    @pytest.mark.skipif(six.PY2,
+        reason="urrlib has incompatible Py2 API, but we will deprecate anyway")
+    def test_open_from_url(self):
+        import urllib.request
+        file_url = "file:///" + self.data('test0.fits')
+        with urllib.request.urlopen(file_url) as urlobj:
+            with fits.open(urlobj) as fits_handle:
+                pass
+
+        # It will not be possible to write to a file that is from a URL object
+        for mode in ('ostream', 'append', 'update'):
+            with pytest.raises(ValueError):
+                with urllib.request.urlopen(file_url) as urlobj:
+                    with fits.open(urlobj, mode=mode) as fits_handle:
+                        pass
+
+    @remote_data(source='astropy')
+    def test_open_from_remote_url(self):
+        import urllib.request
+        remote_url = 'https://astropy.stsci.edu/data/allsky/allsky_rosat.fits'
```

Comment:
I will push an update using `dataurl` to get the base URL path. I might argue in favor of keeping the tests separate since they test fairly different code paths (as you discovered when you tried to test a remote URL). However I'll defer to your judgment.